src/main/java/org/candlepin/subscriptions/subscription/SubscriptionTableController.java

```diff
@@ -0,0 +1,246 @@
+/*
+ * Copyright Red Hat, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * Red Hat trademarks are not licensed under GPLv3. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+package org.candlepin.subscriptions.subscription;
+
+import static org.candlepin.subscriptions.resource.ResourceUtils.*;
+
+import java.time.OffsetDateTime;
+import java.util.*;
+import javax.validation.constraints.Min;
+import org.candlepin.subscriptions.db.SubscriptionCapacityViewRepository;
+import org.candlepin.subscriptions.db.model.ServiceLevel;
+import org.candlepin.subscriptions.db.model.SubscriptionCapacityView;
+import org.candlepin.subscriptions.db.model.Usage;
+import org.candlepin.subscriptions.resource.ResourceUtils;
+import org.candlepin.subscriptions.util.ApplicationClock;
+import org.candlepin.subscriptions.utilization.api.model.*;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Component;
+
+@Component
+public class SubscriptionTableController {
+
+  private final SubscriptionCapacityViewRepository subscriptionCapacityViewRepository;
+  private final ApplicationClock clock;
+
+  SubscriptionTableController(
+      SubscriptionCapacityViewRepository subscriptionCapacityViewRepository,
+      ApplicationClock clock) {
+    this.subscriptionCapacityViewRepository = subscriptionCapacityViewRepository;
+    this.clock = clock;
+  }
+
+  public SkuCapacityReport capacityReportBySku( // NOSONAR
+      ProductId productId,
+      @Min(0) Integer offset,
+      @Min(1) Integer limit,
+      ServiceLevelType sla,
+      UsageType usage,
+      Uom uom,
+      SkuCapacityReportSort sort,
+      SortDirection dir) {
+    /*
+    Notes:
+    - Ignoring beginning and ending query params, as the table should (currently, anyway) only
+      report on the latest subscription status information.
+    - The implementation will report "subscription end" events for Next Event ONLY. That is,
+      future-dated subs are not taken into account when reporting the Next Event; only currently
+      active subs are used for calculating Next Event.
+    */
+
+    OffsetDateTime reportEnd = clock.now();
+    OffsetDateTime reportStart = reportEnd.minusYears(1);
```

Comment:
This will (unintentionally) exclude subs it shouldn't. This should actually be `clock.now()` as well.

Reply:
By that do you mean that `reportStart = reportEnd` so that the variables aren't holding two slightly different times?  Also, do we need/want to have an end date and a start date if we always want to use the current date?

Reply:
@lindseyburnett, we anticipate supporting the date range at some point, and yeah, I mean `reportStart = reportEnd`.

Reply:
@kahowell I have made this change for now, but I do have an open question about the 1 year duration requirement that Dan said we needed to do. I can make that update in a different PR based on the outcome of my discussion with him. Let me know if that works for you.